yb-voyager/cmd/live_migration.go

```diff
@@ -205,23 +210,12 @@ func processEvents(chanNo int, evChan chan *tgtdb.Event, lastAppliedVsn int64, d
 		}
 
 		start := time.Now()
-		var numInserts, numUpdates, numDeletes int64
-		numInserts, numUpdates, numDeletes = 0, 0, 0
-		for _, event := range batch {
-			switch event.Op {
-			case "c":
-				numInserts++
-			case "u":
-				numUpdates++
-			case "d":
-				numDeletes++
-			}
-		}
-		err := tdb.ExecuteBatch(migrationUUID, tgtdb.EventBatch{Events: batch, ChanNo: chanNo, NumInserts: numInserts, NumUpdates: numUpdates, NumDeletes: numDeletes})
+		eventBatch := &tgtdb.EventBatch{Events: batch, ChanNo: chanNo, EventCounts : &tgtdb.EventCounter{}, EventCountsByTable: make(map[string]*tgtdb.EventCounter)}
```

Comment:
Define a new function `eventBatch := tgtdb.NewEventBatch(batch, chanNo)`. The function internally iterates through the `batch` and correctly initialises `EventCounts` and `EventCountsByTable`. With that, you will always have correct batch stats already computed at the time of creation.